### PR TITLE
art-2519 add check upstreampulls function

### PR DESCRIPTION
**Pull request by Ximinhan**
add check imagestreams prs list function in doozer used by image-health check job to collect all opened pr of a target release.

### CODE REVIEW

File: doozerlib/cli/images_upstreampulls.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import yaml
+
+# external
+from github import Github
+
+# doozerlib
+from doozerlib import constants
+from doozerlib.cli import cli, pass_runtime
+
+github_token = os.getenv(constants.GIT_TOKEN)
```
Comment (vfreex) [inline doozerlib/cli/images_upstreampulls.py:12]: Git != GitHub. There is no such thing called `GIT_TOKEN`. Would like to rename it to something like `GITHUB_TOKEN`.

File: doozerlib/cli/images_upstreampulls.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+github_client = Github(github_token)
+repo = github_client.get_repo("openshift/ocp-build-data")
+
+
+@cli.command("images:upstreampulls", short_help="Check upstream PRs in open status")
```
Comment (vfreex) [inline doozerlib/cli/images_upstreampulls.py:17]: It doesn't seem to me that the verb name `images:upstreampulls` or the help message explains well about what it  intends to do. Would like to have a better verb name like `images:streams prs list` for consistency and improved help message.

File: doozerlib/cli/images_upstreampulls.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+# external
+from github import Github
+
+# doozerlib
+from doozerlib import constants
+from doozerlib.cli import cli, pass_runtime
+
+github_token = os.getenv(constants.GIT_TOKEN)
+github_client = Github(github_token)
+repo = github_client.get_repo("openshift/ocp-build-data")
```
Comment (vfreex) [inline doozerlib/cli/images_upstreampulls.py:14]: `openshift/ocp-build-data` shouldn't be hardcoded here. How could I run the doozer with my own fork of ocp-build-data?

Global variables should be avoided as long as you can.

File: doozerlib/cli/images_upstreampulls.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        if 'content' not in ydata:
+            continue
+        if 'git' not in ydata['content']['source']:
+            continue
+        upstream_project = ydata['content']['source']['git']['url'].split(':')[-1].split('/')[0]
+        upstream = ydata['content']['source']['git']['url'].split('/')[-1].split('.')[0]
+        if upstream_project == "openshift-priv":
+            xrepo = github_client.get_repo("openshift/" + upstream)
+        else:
+            xrepo = github_client.get_repo(upstream_project + "/" + upstream)
```
Comment (vfreex) [inline doozerlib/cli/images_upstreampulls.py:35]: See https://github.com/openshift/doozer/blob/c7e31761b3ec312d4c3101bd3c93c94fc53c5025/doozerlib/cli/images_streams.py#L650 for how to correctly resolve the upstream repo url and branch for each image.

File: doozerlib/cli/images_upstreampulls.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        pulls = xrepo.get_pulls(state='open', sort='created')
+        for pr in pulls:
+            if pr.user.login == "openshift-bot" and pr.base.ref == "release-" + version:
```
Comment (vfreex) [inline doozerlib/cli/images_upstreampulls.py:38]: Please don't hardcode the login name and branch name.
login should be configurable, e.g. via command line option, and the branch name is not always follow `release-*`. There are exceptions.

File: doozerlib/cli/images_upstreampulls.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                            pr.html_url, pr.created_at))
+    # print dict data
+    for key, val in retdata.items():
+        if len(val) == 0:
+            continue
+        print(">[{}]".format(key))
+        for img, prs in val.items():
+            print(" -[{}]".format(img))
+            for pr in prs:
+                print("   {}".format(pr))
```
Comment (vfreex) [inline doozerlib/cli/images_upstreampulls.py:62]: Use `yaml.dump`

**Review by vfreex — CHANGES_REQUESTED** [review]

File: doozerlib/cli/images_streams.py [hunk 7]
```diff
@@ -530,6 +530,59 @@ def prs():
...
+                            pr.html_url, pr.created_at))
+    # format output
+    for key, val in retdata.items():
+        if len(val) == 0:
+            continue
+        print(">[{}]".format(key))
+        for img, prs in val.items():
+            print(" -[{}]".format(img))
+            for pr in prs:
+                print("   {}".format(pr))
```
Comment (sosiouxme) [inline doozerlib/cli/images_streams.py:583]: formatting yaml yourself is a recipe for disaster, how about just using the `yaml` module...

**Review by sosiouxme — CHANGES_REQUESTED** [review]

**Comment by sosiouxme** [comment]
> ### Build #9
> ```
> Failed to build pygit2
> ERROR: Could not build wheels for pygit2 which use PEP 517 and cannot be installed directly
> ```

`git fetch && git rebase origin/master` should fix that up.

File: doozerlib/cli/images_streams.py [hunk 8]
```diff
@@ -530,6 +532,35 @@ def prs():
     pass
 
 
+@prs.command('list', short_help='List all open prs for upstream repo, run this need GITHUB_TOKEN var from env got set.')
```
Comment (sosiouxme) [inline doozerlib/cli/images_streams.py:535]: I would reword a little bit shorter:
```suggestion
@prs.command('list', short_help='List all open prs for upstream repos (requires GITHUB_TOKEN env var to be set.')
```

**Review by sosiouxme — APPROVED** [review]
looks good, will merge with one little change
